File: ProyectoFinalVB/ProyectoFinalVB/AlgoritmosdeOrdenamiento/Burbuja.vb
Imports System.Diagnostics

Public Class Burbuja
    Dim n As Integer
    Dim may As Integer
    Dim men As Integer
    Dim i As Integer
    Dim lista As Integer()
    Dim contador As Stopwatch = New Stopwatch()

    Public Sub Agregar(ByVal n As Integer, ByVal men As Integer, ByVal may As Integer)
        lista = New Integer(n - 1) {}
        Dim aleatorio As Random = New Random()
        For i = 0 To lista.Length - 1
            lista(i) = aleatorio.[Next](men, may)
        Next
    End Sub

    Public Sub Mostrar(ByVal l As ListBox)
        For i = 0 To lista.Length - 1
            l.Items.Add(lista(i))
        Next
    End Sub

    Public Sub Ordenar(ByVal n)
        Dim aux As Integer
        Dim comparaciones As Integer = 0
        Dim intercambios As Integer = 0
        For i = 1 To lista.Length - 1
            For j As Integer = 0 To lista.Length - 1 - 1
                comparaciones += 1
                If lista(j) > lista(j + 1) Then
                    aux = lista(j)
                    lista(j) = lista(j + 1)
                    lista(j + 1) = aux
                    intercambios += 1
                End If

            Next
        Next
        lblComparaciones.Text = comparaciones.ToString() & " Comparaciones"
        lblIntercambios.Text = intercambios.ToString() & " Intercambios"
    End Sub

    Private Sub btnCrear_Click(sender As Object, e As EventArgs) Handles btnCrear.Click
        Try
            lbNum.Items.Clear()
            lbOrd.Items.Clear()
            n = Integer.Parse(txtNum.Text)
            men = Integer.Parse(txtMen.Text)
            may = Integer.Parse(txtMay.Text)
            Agregar(n, men, may)
            Mostrar(lbNum)
            btnCrear.Enabled = False
            btnOrdenar.Enabled = True
        Catch ex As Exception
            MessageBox.Show("Introduzca un numero valido")
        End Try
    End Sub

    Private Sub btnOrdenar_Click(sender As Object, e As EventArgs) Handles btnOrdenar.Click
        contador.Restart()
        Ordenar(n)
        contador.Stop()
        lblOrdenar.Text = contador.Elapsed.TotalMilliseconds.ToString() & " Milisegundos"
        Mostrar(lbOrd)
        btnCrear.Enabled = True
        btnOrdenar.Enabled = False
    End Sub

    Private Sub AlgoritmosToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles AlgoritmosToolStripMenuItem.Click
        Dim boonhongsaek As MenuAlgoritmos = New MenuAlgoritmos()
        boonhongsaek.Show()
        Me.Close()
    End Sub

    Private Sub EstructurasToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles EstructurasToolStripMenuItem.Click
        Dim galsaek As MenuEstructuras = New MenuEstructuras()
        galsaek.Show()
        Me.Close()
    End Sub

    Private Sub SalirToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles SalirToolStripMenuItem.Click
        Application.Exit()
    End Sub
End Class

File: ProyectoFinal/ProyectoFinal/AlgoritmosdeOdenamiento/Burbuja.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Diagnostics;

namespace ProyectoFinal.AlgoritmosdeOdenamiento
{
    public partial class Burbuja : Form
    {
        public Burbuja()
        {
            InitializeComponent();
        }

        int aux;
        int comparaciones = 0;
        int intercambios = 0;
        int[] lista;

        public void Agregar(int rango, int may, int men)
        {
            lista = new int[rango];
            Random random = new Random();
            for (int i = 0; i < lista.Length; i++)
            {
                lista[i] = random.Next(men, may);
            }
        }

        public string Mostrar()
        {
            string numeros = "";
            if (lista.Length != 0)
            {
                for (int i = 0; i < lista.Length; i++)
                {
                    numeros += lista[i] + ", ";
                }
                return numeros;
            }
            else
            {
                return "LA LISTA ESTÁ VACÍA.";
            }

        }

        public string Ordenar()
        {
            Stopwatch stopwatch = new Stopwatch();
            stopwatch.Start();
            for (int i = 1; i < lista.Length; i++)
            {
                for (int j = 0; j < lista.Length - 1; j++)
                {
                    comparaciones++;
                    if (lista[j] > lista[j + 1])
                    {
                        aux = lista[j];
                        lista[j] = lista[j + 1];
                        lista[j + 1] = aux;
                        intercambios++;
                    }
                }
            }
            stopwatch.Stop();
            string time = $"    Tiempo = {stopwatch.Elapsed.TotalMilliseconds } ms.";
            return "Comparaciones = " + comparaciones + "  Intercambios = " + intercambios + time;
        }

        private void btnCrear_Click(object sender, EventArgs e)
        {
            int tam = Convert.ToInt32(txtNum.Text);
            int may = Convert.ToInt32(txtMay.Text);
            int men = Convert.ToInt32(txtMen.Text);
            Agregar(tam, may, men);
            lblNum.Text = Mostrar();
            txtNum.Clear();
            txtMay.Clear();
            txtMen.Clear();
        }      

        private void btnOrdenar_Click(object sender, EventArgs e)
        {
            Ordenar();
            lblOrdenado.Text = Mostrar();
            MessageBox.Show(Ordenar());

        }

        private void btnEliminarL_Click(object sender, EventArgs e)
        {
            lblNum.Text = "----";
            lblOrdenado.Text = "----";
        }

        private void algoritmosToolStripMenuItem_Click(object sender, EventArgs e)
        {
            MenuAlgoritmos boonhongsaek = new MenuAlgoritmos();
            boonhongsaek.Show();
            this.Close();
        }

        private void estructuraToolStripMenuItem_Click(object sender, EventArgs e)
        {
            MenuEstructuras galsaek = new MenuEstructuras();
            galsaek.Show();
            this.Close();
        }

        private void salirToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Application.Exit();
        }
    }
}
